lib/checklists/change_note.rb

```diff
@@ -1,23 +1,37 @@
 class Checklists::ChangeNote
+  include ActiveModel::Validations
+
   CONFIG_PATH = Rails.root.join('lib/checklists/change_notes.yaml')
 
+  validates_presence_of :action_id
+  validates_inclusion_of :type, in: %w(addition content_change)
+  validates_format_of :time, with: /\d{4}-\d{2}-\d{2} \d{2}:\d{2}/
+  validates_presence_of :note, if: -> { type == "content_change" }
+
+  validate do
+    errors.add("ID not a UUID") unless id.length == SecureRandom.uuid.length
+  end
```

Comment:
```suggestion
validates_length_of :id, is: SecureRandom.uuid.length, message: "ID not a UUID"
```

Reply:
Or should this be a:
`validates_format_of?`

Reply:
Or:
 `validate { errors.add("ID not a UUID") unless UUID.validate(id) }`

Reply:
Nice - updated 👍 .